packages/headless/src/controllers/triggers/headless-notify-trigger.ts

```diff
@@ -0,0 +1,76 @@
+import {SearchEngine} from '../../app/search-engine/search-engine';
+import {TriggerSection} from '../../state/state-sections';
+import {triggers} from '../../app/reducers';
+import {buildController, Controller} from '../controller/headless-controller';
+import {loadReducerError} from '../../utils/errors';
+import {logNotifyTrigger} from '../../features/triggers/trigger-analytics-actions';
+
+/**
+ * The `NotifyTrigger` controller handles Notify triggers.
+ */
+export interface NotifyTrigger extends Controller {
+  /**
+   * the state of the `NotifyTrigger` controller.
+   */
+  state: NotifyTriggerState;
+}
+
+/**
+ * A scoped and simplified part of the headless state that is relevant to the `NotifyTrigger` controller.
+ */
+export interface NotifyTriggerState {
+  /**
+   * The notification to present to the user after receiving a notification trigger.
+   */
+  notification: string;
+}
+
+/**
+ * Creates a `NotifyTrigger` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @returns A `NotifyTrigger` controller instance.
+ * */
+export function buildNotifyTrigger(engine: SearchEngine): NotifyTrigger {
+  if (!loadNotifyTriggerReducers(engine)) {
+    throw loadReducerError;
+  }
+
+  const controller = buildController(engine);
+  const {dispatch} = engine;
+
+  const getState = () => engine.state;
+
+  let previousNotify: string = getState().triggers.notification;
+
+  return {
+    ...controller,
+
+    subscribe(listener: () => void) {
+      const strictListener = () => {
+        const hasChanged = previousNotify !== this.state.notification;
+        previousNotify = this.state.notification;
+
+        if (hasChanged && getState().triggers.notification) {
```

Comment:
If use this.state.notification to be consistent

Reply:
I think Sami mentioned something about the user potentially overriding the controller's get state method. But if we refer to the engine state directly instead we can prevent any potential overriding from hindering the controller's intended functionality.